src/MirrorNode.sol

```diff
@@ -51,13 +52,36 @@ abstract contract MirrorNode {
         return 0;
     }
 
+    function getAccountAddress(string memory accountId) external returns (address) {
+        if (bytes(accountId).length == 0
+            || keccak256(bytes(accountId)) == keccak256(bytes("null"))
+            || keccak256(abi.encodePacked(accountId)) == keccak256(abi.encodePacked(bytes32(0)))) {
```

Comment:
question: all these different variants come from the Mirror Node?

Reply:
Nope, `parseJsonString` resolves to `"null"` when the field in the json has a `null` value, other checks are just safety checks to confirm that the given address is valid before calling mirror node